rllib/evaluation/worker_set.py

```diff
@@ -610,6 +664,30 @@ def valid_module(class_path):
 
         return worker
 
+    def _worker_health_check(self) -> List[int]:
+        """Performs a health-check on each remote worker.
+
+        Returns:
+            List of indices (into `self._remote_workers` list) of faulty workers.
+            Note that index=1 is the 0th item in `self._remote_workers`.
+        """
+        logger.info("Health checking all workers ...")
+        checks = []
+        for worker in self.remote_workers():
+            _, obj_ref = worker.sample_with_count.remote()
```

Comment:
I understand this is how we have been doing this, so maybe just a TODO suggestion.
usually there will be a separate health check function that performs actual health checks (able to respond to probes, all required data available, etc). and it is usually very light weight, so it doesn't impact actual workload.
I don't know if getting an actual SampleBatch of data then dropping it is the best kind of health probe (it could be really expensive), so this suggestion.

Reply:
Hmm, I actually think this is fine. `sample()` (or `sample_with_count()`) is the most important method to run and we only perform this check, if we get any Ray errors during the training iterations.

In other words: "able to respond to probes, all required data available, etc": I feel like that's exactly what `sample()` does. I agree we should probably check the returned SampleBatch data.

Reply:
ok, ultimately your call. I guess we only probe when there is a worker failure, so it doesn't happen often and should be ok. if it's a periodic probe, we probably should make this much lighter.

Reply:
Agreed. Then again, not sure what to use (in today's RolloutWorker) that would serve as a probing tool, other than `sample()`. I'll add a TODO.